src/js/navIcons.js

```diff
@@ -0,0 +1,106 @@
+/*
+
+Copyright 2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+(function ($, fluid) {
+
+    "use strict";
+
+    fluid.registerNamespace("gpii.firstDiscovery");
+
+    /*
+     * The nav icon
+     */
+    fluid.defaults("gpii.firstDiscovery.icon", {
+        gradeNames: ["fluid.viewComponent", "autoInit"],
+        position: null,  // must be supplied by integrators
+        selectors: {
+            activeIndicator: ".gpiic-fd-activeIndicator",
+            doneIndicator: ".gpiic-fd-doneIndicator"
+        },
+        styles: {
+            active: "gpii-fd-active",
+            show: "gpii-fd-show"
+        },
+        modelListeners: {
+            "isActive.setState": {
+                listener: "gpii.firstDiscovery.icon.setState",
+                args: ["{that}", "{change}.value", "{change}.oldValue"]
+            }
+        }
+    });
+
+    gpii.firstDiscovery.icon.setState = function (that, isActive, isActivePrev) {
+        var activeCss = that.options.styles.active,
+            showCss = that.options.styles.show,
+            activeIndicator = that.locate("activeIndicator");
+
+        that.container.toggleClass(activeCss, isActive);
+        activeIndicator.toggleClass(showCss, isActive);
+
+        if (isActivePrev && !isActive) {
+            that.locate("doneIndicator").addClass(that.options.styles.show);
```

Comment:
i also thought about that. I didn't continue to use isVisited model state is 'cuz it's really necessary at this point. Using that is basically split these 3 lines of code into one change request and another model listener, which seems indirect and unnecessary. I think we can make that change if someday isVisited needs to be listened by more than one sources.